test/suites-ostree/upgrade-ostree.robot

```diff
@@ -0,0 +1,77 @@
+*** Settings ***
+Documentation       Tests related to MicroShift upgradeability on OSTree-based systems
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-process.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+
+*** Test Cases ***
+Rebooting Healthy System Should Result In Data Backup
+    [Documentation]    Check if rebooting healthy system will result in backing up of MicroShift data
+
+    Wait Until Greenboot Health Check Exited
+    System Should Be Healthy
+    Remove Existing Backup For Current Deployment
+
+    Reboot MicroShift Host
+    Wait For MicroShift Service
+
+    Backup For Booted Deployment Should Exist
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Wait Until Greenboot Health Check Exited
+    [Documentation]    Wait until greenboot healthchecks are done
+
+    Wait Until Keyword Succeeds    5m    15s
+    ...    Greenboot Health Check Exited
+
+Greenboot Health Check Exited
+    [Documentation]    Checks if greenboot-healthcheck finished running
+
+    ${value}=    Get Systemd Setting    greenboot-healthcheck.service    SubState
+    Should Be Equal As Strings    ${value}    exited
+
+Remove Existing Backup For Current Deployment
+    [Documentation]    Remove backups for current deployment
+
+    ${path}=    Get Booted Deployment Backup Prefix Path
+
```

Comment:
Should we test path here with 'Should Not Be Empty'? I see that 'Get Booted Deployment Backup Prefix Path' has a check for empty values. However, if the code is changed to use a different key word at some point to get path we would not want to accidentally call 'rm -rf *'